src/device-manager/java/src/nl/Weave/DataManagement/GenericTraitUpdatableDataSink.java

```diff
@@ -0,0 +1,74 @@
+/*
+
+    Copyright (c) 2019 Google, LLC.
+    All rights reserved.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+ */
+
+
+package nl.Weave.DataManagement;
+
+import android.os.Build;
+import android.util.Log;
+
+import java.math.BigInteger;
+import java.util.EnumSet;
+import java.util.Random;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Iterator;
+
+public interface GenericTraitUpdatableDataSink
+{
+    public void setSigned(String path, int value, boolean isConditional);
+    public void setSigned(String path, long value, boolean isConditional);
+    public void setSigned(String path, BigInteger value, boolean isConditional);
+    public void setUnsigned(String path, int value, boolean isConditional);
+    public void setUnsigned(String path, long value, boolean isConditional);
+    public void setUnsigned(String path, BigInteger value, boolean isConditional);
+    public void set(String path, double value, boolean isConditional);
+    public void set(String path, boolean value, boolean isConditional);
+    public void set(String path, String value, boolean isConditional);
+    public void set(String path, byte[] value, boolean isConditional);
+    public void setNULL(String path, boolean isConditional);
+    public void setSigned(String path, int value);
+    public void setSigned(String path, long value);
+    public void setSigned(String path, BigInteger value);
+    public void setUnsigned(String path, int value);
+    public void setUnsigned(String path, long value);
+    public void setUnsigned(String path, BigInteger value);
+    public void set(String path, double value);
+    public void set(String path, boolean value);
+    public void set(String path, String value);
+    public void set(String path, byte[] value);
+    public void setNULL(String path);
+    public int getInt(String path);
+    public long getLong(String path);
+    public BigInteger getBigInteger(String path, int bitLen);
```

Comment:
For the BigInteger/String/byte[] accessors, is it expected that these will return null if the path is invalid, or it will throw an exception? If null, can these be annotated with @Nullable?

Reply:
@nullable is from javax, which is not generic? I cannot use javax when compiling weave.

in jni layer, for example, when getLong failed, it will ThrowError
    if (err != WEAVE_NO_ERROR && err != WDM_JNI_ERROR_EXCEPTION_THROWN)
    {
        ThrowError(env, err);
    }

void ThrowError(JNIEnv *env, WEAVE_ERROR errToThrow)
{
    WEAVE_ERROR err = WEAVE_NO_ERROR;
    jthrowable ex;

    err = N2J_Error(env, errToThrow, ex);
    if (err == WEAVE_NO_ERROR)
        env->Throw(ex);
}

Reply:
@jaylogue any ideas if we can add the jsr305 lib as a dependency easily? This would be very helpful for our use from Kotlin code in the Nest/Home apps.